### PR TITLE
Shopify to Salesforce - Customers

### PR DESCRIPTION
**Pull request by FrendsSarlinS**
#98 

Checklist for review:
- [x] Process name is fine, uses normal casing and descriptive
- [x] Process description is fine, uses normal casing, dots and is descriptive
  - [x] Element names use normal casing, no unneeded capital letters.
- [x] Correct casing used for formats and terms, e.g. CSV, FTP, JSON, Avro
  - [x] Element names do not contain dots
  - [x] Comments / note element text are in good language and end with dots.
- [x] Template parameters checked
  - [x] Parameter descriptions are good and end with dots.
  - [x] Sensitive parameters are marked as secrets
- [x] "Turn off logging" switch is on for any task that has sensitive data as output
- [x] End "balls" like error throws and return statements have titles (double-click) and reasonable information.
  - [x] Results have good summaries, preferably with statistics of some sort like "processed X files"
  - [x] Errors have some error info, usually from task (e.g. some ErrorMessage property contains additional error info)
- [x] Process is organised with groups (dashed boxes)
- [x] Straight lines are preferred where possible
- [x] Subprocesses are forbidden
- [x] Text mode is preferred over Expressions in expressions, e.g. instead of *$"This is {#var.error}"* use *This is {{#var.error}}*
- [x] Process metadata file is present and content is fine (version number and name)
- [x] Process export file is present (process.json)
- [x] Process description file is present (all sections filled in)
- [x] Process description has a SVG screenshot of the process
- [x] Tempalte files are going to the correct folder

## Summary by CodeRabbit

- **New Features**
  - Introduced a new template for synchronizing customer data from Shopify to Salesforce, ensuring seamless integration between the two platforms.
  - The template includes details on prerequisites, implementation, usage notes, error handling, and managing deleted customers.
  - Added metadata for the transformation process, specifying version and systems involved.

### CODE REVIEW

**Review by jannejjj — COMMENTED** [review]
Looks good in general, some comments:

1. The returns inside the for-loop don't have titles, and the returns/throws use expressions instead of text (the final return needs to use expression because it has a condition check).

2. Since the customers are inserted into Salesforce as Contact objects, we should probably change the wording in the process to match that. For example, "Get customer information from Salesforce" could be "Get contact information from Salesforce.

**Comment by FrendsSarlinS** [comment]
New commit is in!

1) Titles were added to the returns and the throw output is now Text instead of Expression. In returns, this could not be implemented because of the exceptions variable.

2) The wording was changed in the Salesforce task, and the customer-contact relation is now noted in the group annotation and the exclusive decision. For the rest of the tasks, I thought it would be best to keep the source format, i.e. customer.

**Comment by jannejjj** [comment]
I didn't approve this yet, since I'm wondering if it would be clearer to speak of contacts in the insert/update tasks instead of customers, since the customers from shopify are being inserted to salesforce as contacts. Otherwise everything seems good! 👍

**Review by jannejjj — APPROVED** [review]
With the latest changes, everything looks good.

File: Templates/E-Commerce/Shopify/Shopify to Salesforce - Customers/long-description.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+This template fetches all the customers from Shopify, and then either inserts or updates them as contacts in Salesforce. To determine if a Shopify customer already has a corresponding contact in Salesforce, the email addresses of the Shopify customers are compared to those of the existing contacts in Salesforce.
+
+Customers that have been deleted from Shopify are not transferred to Salesforce.
+
+Process variables include the credentials and URLs for connecting to both Shopify and Salesforce.
+
+# Error Handling
+
+There is an error check after any tasks regarding Shopify and Salesforce. If an error occurs during customer handling, an error message is added to the exception variable. Then the next contact is taken for handling. All error messages are shown at the end of the process.
+Transient errors are not handled separately. Retries for any tasks can be enabled if transient errors are expected.
```
Comment (jefim) [inline Templates/E-Commerce/Shopify/Shopify to Salesforce - Customers/long-description.md:24]: double newline

File: Templates/E-Commerce/Shopify/Shopify to Salesforce - Customers/long-description.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+This template synchronizes customers from Shopify to Salesforce as contacts.
+Using this template, you can synchronize the list of customers and their names, emails, and addresses from Shopify to Salesforce.
```
Comment (jefim) [inline Templates/E-Commerce/Shopify/Shopify to Salesforce - Customers/long-description.md:2]: double newline

**Review by jefim — CHANGES_REQUESTED** [review]

File: Templates/E-Commerce/Shopify/Shopify to Salesforce - Customers/metadata.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+    "name": "Shopify to Salesforce - Customers",
+    "version": "1.0.0",
+    "tags": [],
+    "additionalSystemsUsed": [
```
Comment (jannejjj) [inline Templates/E-Commerce/Shopify/Shopify to Salesforce - Customers/metadata.json:5]: I guess we need to have Shopify in the additional systems 😄

**Review by jannejjj — APPROVED** [review]
The newlines have been added and a mention of Shopify added to the metadata.

**Pull request merged**